api/src/main/java/com/epam/esm/controller/GiftController.java

```diff
@@ -0,0 +1,94 @@
+package com.epam.esm.controller;
+
+import com.epam.esm.model.CustomSearchRequest;
+import com.epam.esm.model.dto.GiftCertificateDto;
+import com.epam.esm.service.GiftService;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiParam;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@RestController
+@RequestMapping("/api/v1/gifts")
+public class GiftController {
+    private final GiftService giftService;
+    private final CustomSearchRequest defaultCustomSearchRequest;
+
+
+    @Autowired
+    public GiftController(GiftService giftService) {
+        this.giftService = giftService;
+        defaultCustomSearchRequest = CustomSearchRequest.builder().build();
+    }
+
+
+    @GetMapping
+    @ApiOperation(value = "Api v1. Get all gifts")
+    public ResponseEntity<List<GiftCertificateDto>> allGifts(
+            @ApiParam(name = "descriptionPrefix", value = "Search by gift description") String descriptionPrefix,
+            @ApiParam(name = "namePrefix", value = "Search by gift name") String namePrefix,
+            @ApiParam(name = "sortField", value = "Sort by fields \"name\" or \"date\"") String sortField,
+            @ApiParam(name = "sortMethod", value = "Sort methods \"asc\" or \"desc\"") String sortMethod,
+            @ApiParam(name = "tagNamePrefix", value = "Search by tag name") String tagNamePrefix
+            ) {
+        List<GiftCertificateDto> allGifts;
+
+        CustomSearchRequest customSearchRequest = CustomSearchRequest.builder()
+                .descriptionPrefix(descriptionPrefix)
+                .namePrefix(namePrefix)
+                .sortField(sortField)
+                .sortMethod(sortMethod)
+                .tagNamePrefix(tagNamePrefix).build();
+
+        allGifts = !customSearchRequest.equals(defaultCustomSearchRequest) ? giftService.searchGifts(customSearchRequest) : giftService.getAllGifts();
```

Comment:
it is easier to read this line if you avoid negative ! in condition and rewrite it in the following way
` List<GiftCertificateDto> allGifts = customSearchRequest.equals(defaultCustomSearchRequest) 
                                           `? giftService.searchGifts(customSearchRequest)`
                                           ` : giftService.getAllGifts();`